### PR TITLE
Localized game-mode and save-slot-select-ui-handler

### PR DESCRIPTION
**Pull request by josericardo-fo**
## What are the changes?

Localized game-mode and save-slot-select-ui-handler and added ptBR translations
## Why am I doing these changes?

Those pages weren't localized yet
## What did change?

Translations and text in game-mode and save-slot-select-ui-handler files
"Empty" locale was only used in the save-slot-select-ui-handler, so I moved it from menu.ts to save-slot-select-ui-handler.ts
### Before
image
### After
image

### Before
image
image
### After
image
image

### Before
image
### After
image

## How to test the changes?

Check the new game screen, load game screen and the text when overwriting a save slot.
## Checklist
- [ ] There is no overlap with another PR?
Overlap with #1674, but he just localized the overwriteData text
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I tested the changes (manually)?
    - [X] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by flx-sta** [comment]
There is a lot of “noise” in the PR because you added the English translation for every locale.
Did you try if it will fall-back to English by default ? Because this makes reviewing more of a chore and it is Is intact not a translation. Just noise

**Comment by josericardo-fo** [comment]
What do you mean? When I localize something, I don't translate the localized files beacuse I'm not qualified to do that. It's up to the other devs to add PRs to my PR to translate or wait for it to get merged than create their own PRs.

**Comment by flx-sta** [comment]
I mean something like the `src/locales/de/game-mode.ts` file. It now contains the English translations. But if English is the fallback language anyways that shouldn’t be necessary and those are not “translations” anyway.

**Comment by josericardo-fo** [comment]
I agree. But, as I said, I localized it and now other people can translate it if they want to. If it is working in every language, than its fine.

**Comment by flx-sta** [comment]
That’s why I’m saying: see what is being printed if you remove a language that is not translated. Technically it should show the English translation. This you don’t have to provide those inside the locale file of a different language

File: src/locales/de/game-mode.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { SimpleTranslationEntries } from "#app/plugins/i18n";
+
+export const gameMode: SimpleTranslationEntries = {
+  "classic": "Classic",
+  "endless": "Endless",
+  "endlessSpliced": "Endless (Spliced)",
+  "dailyRun": "Daily Run",
+  "unknown": "Unknown",
```
Comment (CodeTappert) [inline src/locales/de/game-mode.ts:8]:
```suggestion
  "classic": "Klassik",
  "endless": "Endlos",
  "endlessSpliced": "Endlos (Fusion)",
  "dailyRun": "Täglicher Run",
  "unknown": "Unbekannt",
```

File: src/locales/de/save-slot-select-ui-handler.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { SimpleTranslationEntries } from "#app/plugins/i18n";
+
+export const saveSlotSelectUiHandler: SimpleTranslationEntries = {
+  "overwriteData": "Overwrite the data in the selected slot?",
+  "loading": "Loading...",
+  "wave": "Wave",
+  "lv": "Lv",
+  "empty": "Leer",
```
Comment (CodeTappert) [inline src/locales/de/save-slot-select-ui-handler.ts:8]:
```suggestion
  "overwriteData": "Den ausgewählten Speicherstand überschreiben?",
  "loading": "Läd...",
  "wave": "Welle",
  "lv": "Lvl",
  "empty": "Leer",
```

File: src/locales/es/game-mode.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { SimpleTranslationEntries } from "#app/plugins/i18n";
+
+export const gameMode: SimpleTranslationEntries = {
```
Comment (GoldTra) [inline src/locales/es/game-mode.ts:3]:
```suggestion
export const gameMode: SimpleTranslationEntries = {
```

**Review by GoldTra — CHANGES_REQUESTED** [review]

File: src/locales/es/save-slot-select-ui-handler.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { SimpleTranslationEntries } from "#app/plugins/i18n";
+
+export const saveSlotSelectUiHandler: SimpleTranslationEntries = {
+  "overwriteData": "Overwrite the data in the selected slot?",
+  "loading": "Loading...",
+  "wave": "Wave",
+  "lv": "Lv",
```
[the remark this answers is not in the document]
Comment (CodeTappert) [reply src/locales/es/save-slot-select-ui-handler.ts:7]: The override thing may be too long. Could someone check that

[the remark this answers is not in the document]
Comment (GoldTra) [reply src/locales/es/save-slot-select-ui-handler.ts:7]: > The override thing may be too long. Could someone check that

Captura1
Captura3

[the remark this answers is not in the document]
Comment (CodeTappert) [reply src/locales/es/save-slot-select-ui-handler.ts:7]: Okay then its all good. Just wanted to be sure since my german sentence was orignal to long

**Comment by Adri1** [comment]
PR opened on your branch with French translations

**Comment by Adri1** [comment]
Does this PR also convet numbers according to the language ?
image
Because I see this and for example in French, we use the comas before deciamals not dots.
It shoud be possible, as it is already the case for many numbers formatting when, for example money is displayed in some places.

**Comment by josericardo-fo** [comment]
I will take a look on that

> Also, does this PR also converts dynamically numbers according to the set language, the navigator or system language? image Because I see this and for example in French, we use comas before deciamals not dots. (and just spaces where in English they use comas) It shoud be possible, as it is already the case for many number formatting when, for example, money is displayed in some places.

**Comment by josericardo-fo** [comment]
@Adri1 
image
At the end, they return a dot by default after decimals. It can be changed, but the comma is kinda strange:
image
I didn't know that there was a way to localize that, looking for it becuse numbers here in Brazil are separated like this "$1.300,50" (one thousand, three hundred dollars and fifty cents) and I didn't insert that in, for example, the MoneyAchvs localization

**Comment by Adri1** [comment]
For example, here is what I did in PR #1656 :
image
I always only encountered `toLocaleString("en-US")` when I happened to come across numbers. So for what you have here, I admit I dont have any more ideas.
But if what I did can give you any ideas to adapt it to your case...

**Comment by josericardo-fo** [comment]
Created a PR for that @Adri1 #1762

**Comment by GoldTra** [comment]
@josericardo-fo PR opened on your branch with Spanish translations

File: src/locales/ko/game-mode.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { SimpleTranslationEntries } from "#app/plugins/i18n";
+
+export const gameMode: SimpleTranslationEntries = {
+  "classic": "Classic",
+  "endless": "Endless",
+  "endlessSpliced": "Endless (Spliced)",
+  "dailyRun": "Daily Run",
+  "unknown": "Unknown",
+  "challenge": "Challenge",
```
Comment (returntoice) [inline src/locales/ko/game-mode.ts:9]:
```suggestion
  "classic": "클래식",
  "endless": "엔들리스",
  "endlessSpliced": "엔들리스(융합체)",
  "dailyRun": "데일리 런",
  "unknown": "언노운",
  "challenge": "챌린지",
```
Korean update

File: src/locales/ko/save-slot-select-ui-handler.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { SimpleTranslationEntries } from "#app/plugins/i18n";
+
+export const saveSlotSelectUiHandler: SimpleTranslationEntries = {
+  "overwriteData": "Overwrite the data in the selected slot?",
+  "loading": "Loading...",
+  "wave": "Wave",
```
Comment (returntoice) [inline src/locales/ko/save-slot-select-ui-handler.ts:6]:
```suggestion
  "overwriteData": "선택한 슬롯에 데이터를 덮어쓰시겠습니까?",
  "loading": "로딩 중...",
  "wave": "웨이브",
```

**Review by returntoice — CHANGES_REQUESTED** [review]
Update for Korean locales

**Review by Tempo-anon — APPROVED** [review]
Going forward let's try having each separate localization go in their own PR since there indeed is a lot of noise to this one which makes it harder to read and review.

**Pull request merged**